benchmarks/sample_consensus/sac_model_cylinder.cpp

```diff
@@ -0,0 +1,70 @@
+#include <pcl/features/normal_3d.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/sample_consensus/method_types.h>
+#include <pcl/sample_consensus/model_types.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/point_types.h>
+
+#include <benchmark/benchmark.h>
+
+static void
+BM_SacModelCylinder(benchmark::State& state, const std::string& file)
+{
+  // All the objects needed
+  pcl::PCDReader reader;
+  pcl::NormalEstimation<pcl::PointXYZ, pcl::Normal> ne;
+  pcl::SACSegmentationFromNormals<pcl::PointXYZ, pcl::Normal> seg;
+
+  // Datasets
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PointCloud<pcl::Normal>::Ptr cloud_normals(new pcl::PointCloud<pcl::Normal>);
+  pcl::ModelCoefficients::Ptr coefficients_cylinder(new pcl::ModelCoefficients);
+  pcl::PointIndices::Ptr inliers_cylinder(new pcl::PointIndices);
+
+  // Read in the cloud data
+  reader.read(file, *cloud);
+
+  // Estimate point normals
+  ne.setInputCloud(cloud);
+  ne.setKSearch(50);
+  ne.compute(*cloud_normals);
+
+  // Create the segmentation object for cylinder segmentation and set all the parameters
+  seg.setOptimizeCoefficients(true);
+  seg.setModelType(pcl::SACMODEL_CYLINDER);
+  seg.setMethodType(pcl::SAC_RANSAC);
+  seg.setNormalDistanceWeight(0.1);
+  seg.setMaxIterations(10000);
```

Comment:
What do you think about adding this?
```suggestion
  seg.setMaxIterations(10000);
  seg.setProbability(1.0);
```

It's going to force RANSAC to do all 10,000 iterations and gives us a better idea about the runtime performance in challenging datasets.